src/resources/js/admin/tickets-commerce-settings.js

```diff
@@ -0,0 +1,282 @@
+	 * Performs an AJAX request to get the partner URL.
+	 *
+	 * @since TBD
+	 *
+	 * @param {String} countryCode The country code.
+	 *
+	 * @return {void}
+	 */
+	obj.requestPartnerUrl = function( countryCode ) {
+		// @todo Add AJAX handler for this.
+		fetch( ajaxurl + `?action=tribe_tickets_paypal_commerce_get_partner_url&countryCode=${ countryCode }` )
+			.then( response => response.json() )
+			.then( function( res ) {
+				// Handle success.
+				if ( true === res.success ) {
+					obj.setupPartnerLink( res.data.partnerLink );
+				}
+
+				obj.buttonState.enable();
+			} )
+			.then( function() {
+				// Handle the error notice.
+				// @todo Add AJAX handler for this.
+				fetch( ajaxurl + '?action=tribe_tickets_paypal_commerce_onboarding_trouble_notice' )
+					.then( response => response.json() )
+					.then( function( res ) {
+						if ( true !== res.success ) {
+							return;
+						}
+
+						function createElementFromHTML( htmlString ) {
+							const div = document.createElement( 'div' );
+							div.innerHTML = htmlString.trim();
+							return div.firstChild;
+						}
+
+						const buttonContainer = document.querySelector( obj.selectors.connectButtonWrap );
+						obj.paypalErrorQuickHelp[0] && obj.paypalErrorQuickHelp.remove();
+						buttonContainer.append( createElementFromHTML( res.data ) );
+					} );
+			} );
+	};
+
+	obj.removeErrors = function() {
+		const errorsContainer = document.querySelector( obj.selectors.errorMessageTemplate );
+
+		if ( errorsContainer ) {
+			errorsContainer.parentElement.remove();
+		}
+	}
+
+	obj.buttonState = {
+		enable: () => {
+			obj.onBoardingButton.attr( 'disabled', false );
+			obj.onBoardingButton.text( obj.onBoardingButton.data( 'initial-label' ) );
+		},
+		disable: () => {
+			// Preserve initial label.
+			if ( ! obj.onBoardingButton.data( 'initial-label' ) ) {
+				obj.onBoardingButton.data( 'initial-label', obj.onBoardingButton.text().trim() );
+			}
+
+			obj.onBoardingButton.attr( 'disabled', true );
+
+			// @todo Replace the i18n text here.
+			obj.onBoardingButton.text( 'Processing text here' );
+		},
+	};
+
+	obj.handleConnectClick = function( evt ) {
+		evt.preventDefault();
+		obj.removeErrors();
+
+		const countryCode = $( obj.selectors.countrySelect ).val();
+
+		obj.buttonState.disable();
+
+		obj.paypalErrorQuickHelp = $( obj.selectors.troubleNotice );
+
+		// Hide paypal quick help message.
+		obj.paypalErrorQuickHelp[0] && obj.paypalErrorQuickHelp.addClass( 'tribe-common-a11y-hidden' );
+
+		obj.requestPartnerUrl( countryCode );
+	};
+
+	obj.handleDisconnectClick = function( evt ) {
+		evt.preventDefault();
+		obj.removeErrors();
+
+		// @todo Show a confirmation modal here.
+		//title: givePayPalCommerce.translations.confirmPaypalAccountDisconnection
+		//desc: givePayPalCommerce.translations.disconnectPayPalAccount
+
+		// On modal confirmation, disconnect the account.
+		$( obj.selectors.connectionSettingContainer ).removeClass( 'tribe-common-a11y-hidden' );
```

Comment:
Let's move `'tribe-common-a11y-hidden'` to the selectors object and use the common `.className()`